File: boj/15. 그리디알고리즘/5585_거스름돈.cpp
// 500엔, 100엔, 50엔, 10엔, 5엔, 1엔
#include <iostream>
using namespace std;

int main() {
	int n, m;
	int cnt=0;
	cin>>n;
	
	m = 1000-n;
	while(m>0){
		if(m>=500){
			m-=500;
			cnt++;
		}else if(m>=100){
			m-=100;
			cnt++;
		}else if(m>=50){
			m-=50;
			cnt++;
		}else if(m>=10){
			m-=10;
			cnt++;
		}else if(m>=5){
			m-=5;
			cnt++;
		}else{
			m-=1;
			cnt++;
		}
	}
	cout<<cnt<<"\n";
	return 0;
}

File: boj/15. 그리디알고리즘/5585_거스름돈.py
n = int(input())
money = 1000 - n
changes = [500, 100, 50, 10, 5, 1]
cnt = 0
i = 0

while money > 0:
	if changes[i] > money:
		i += 1
		continue
	cnt += money // changes[i]
	money %= changes[i]

print(cnt)